### PR TITLE
Fixed LDAP Auth

### PR DESCRIPTION
**Pull request by charlesdhdt**
Fixed LDAP Auth. I realized what I did last week is clearly not good enough and is too specific so I did everything again with a better understanding of ldap

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -96,6 +96,7 @@
         'dev': [
             "Flask>=0.10.1, <0.13",
             "flake8>=3.2.1, <3.4",
+            "mock>=2.0.0, <3.0",
```
Comment (Kriechi) [inline setup.py]: Please use `from unittest import mock` instead of this package.

File: test/mitmproxy/addons/test_proxyauth.py [hunk 2]
```diff
@@ -1,8 +1,8 @@
 import binascii
-import ldap3
 
 import pytest
 
+import mock
```
Comment (Kriechi) [inline test/mitmproxy/addons/test_proxyauth.py]: `from unittest import mock`

**Comment by charlesdhdt** [comment]
I think it is ready to be pushed, what do you think ?

**Comment by Kriechi** [comment]
Thanks! 🎉 👍

**Pull request merged**